curvefs/src/mds/fs_storage.cpp

```diff
@@ -570,5 +620,35 @@ FSStatusCode PersisKVStorage::DeleteFsUsage(const std::string& fsName) {
     return FSStatusCode::OK;
 }
 
+FSStatusCode PersisKVStorage::Tso(uint32_t fsId, uint64_t* ts,
+                                  uint64_t* timestamp) {
+    WriteLockGuard lock(tsLock_);
+    *timestamp = curve::common::TimeUtility::GetTimeofDayMs();
+    auto it = tsMap_.find(fsId);
+    if (it == tsMap_.end()) {
+        *ts = 1;
+    } else {
+        *ts = it->second.ts() + 1;
+    }
+    TS tsInfo;
+    tsInfo.set_ts(*ts);
+    // persist to storage
+    std::string key = codec::EncodeTsKey(fsId);
```

Comment:
Can you use the method of allocating chunkid to allocate ts? so that, you don't have to persist to etch for each ts.